### ISSUE TITLE
Align style of Select components

### ISSUE THREAD

**Issue by ferhoyos** [kiali/kiali#7277]
### What do you want to improve?

Currently there are two types of Select components within Kiali:
- Standard Selects used mainly in Wizards

![image](https://github.com/kiali/kiali/assets/122779323/71d7dfd8-45ff-4f91-bd9e-88af2bb911dd)

- Patternfly Selects used in the rest of the Kiali application:

![image](https://github.com/kiali/kiali/assets/122779323/218f1713-51b4-44fa-b09a-8c4d43844298)

It is a UX good practice to apply the same style for the same type of component, **Select** in this case.

### What is the current behavior?

Two different Select component styles used in Kiali.

### What is the new behavior?

Use the same Select component in Kiali. IMO the best choice is to follow Patternfly's style like the most of the UI elements in Kiali.

**Comment by TomasMatarazzo** [comment]
can i work on this? thx

**Comment by aljesusg** [comment]
Of course.  I assigned this to you ! 

Welcome to Kiali community 
 Thanks @TomasMatarazzo

**Comment by jshaughn** [comment]
@TomasMatarazzo , Thanks!  If you prefer, you can do this incrementally with multiple PRs.  Please feel free to ask for guidance/review.

**Comment by ferhoyos** [comment]
Confirmed with @fkargbo that it is recommended to replace Select components to unify UX style. @TomasMatarazzo, I suggest to create a first PR replacing one of the FormSelect for guidance/review. Then, you can continue with the rest of them. Thanks!

**Comment by amarnath666** [comment]
can i work on this & do i have to change all FormSelect as below image
![Screenshot 2024-06-06 204746](https://github.com/kiali/kiali/assets/145348741/1fc9ee75-0b09-4ba7-a397-7e47c281a7d3)

**Comment by ferhoyos** [comment]
Thanks @amarnath666. The issue consists of replacing all <FormSelect> (https://www.patternfly.org/components/forms/form-select, which embeds browser native select) to <Select> components (https://www.patternfly.org/components/menus/select). There are some examples of <Select> within the Kiali application you can take as reference.

In theory, @TomasMatarazzo is currently working on this issue. Any update from your side?

**Comment by amarnath666** [comment]
Sorry to inform you @ferhoyos , i have learned functional components & in your code there are class components, so i wont be able to contribute

**Comment by jshaughn** [comment]
This issue is back up for grabs...

**Comment by sakshamcodes01** [comment]
> This issue is back up for grabs...

Still up for grabs? :)

**Comment by jshaughn** [comment]
Give it a go, @sakshamcodes01 !

**Comment by besalraj30** [comment]
Is this still up for grabs? @jshaughn

**Comment by jshaughn** [comment]
Hi @besalraj30 , I had assigned it to @sakshamcodes01 but I haven't seen any activity.  So, I'll re-assign to you.  @sakshamcodes01 , please speak up (and open a draft PR) if you have work in progress...

**Comment by besalraj30** [comment]
> Hi @besalraj30 , I had assigned it to @sakshamcodes01 but I haven't seen any activity. So, I'll re-assign to you. @sakshamcodes01 , please speak up (and open a draft PR) if you have work in progress...

Hi @jshaughn, as per the idea proposed by @ferhoyos , all <FormSelect> should bre replaced with <Select>

I'll raise a PR for a single such changes and then follow up with the pending changes, hope thats fine

**Comment by jshaughn** [comment]
@besalraj30 , yes, definitely start with one and ensure all is good with guidance/review from @ferhoyos .

**Comment by besalraj30** [comment]
> @besalraj30 , yes, definitely start with one and ensure all is good with guidance/review from @ferhoyos .

Sure, will raise a PR ASAP

**Comment by besalraj30** [comment]
@jshaughn  @ferhoyos  https://github.com/kiali/kiali/pull/7812

Request you to please review this PR for a single component for the starters.
In case of any changes in this PR, feel free to let me know, will make the changes

**Comment by besalraj30** [comment]
@jshaughn @ferhoyos https://github.com/kiali/kiali/pull/7812

Request you to please review this PR 

i have made the changes as suggested, please verify

**Comment by besalraj30** [comment]
@jshaughn  @ferhoyos https://github.com/kiali/kiali/pull/7828/

Request you to please review this PR

This is for K8sGatewaySelector component, apologies for the delay

**Comment by sakshamcodes01** [comment]
Hi @jshaughn, apologies for the delay. I had faced some personal tragedy recently, so couldn't work on it.
Thank you for your patience though. I will look for other open issues I can start on!

**Comment by jshaughn** [comment]
No problem, @sakshamcodes01, and we are sorry to hear that news.  When you are able we look forward to a contribution.  On this issue, @besalraj30 has been doing a nice job, but there are plenty of other things!

**Comment by ferhoyos** [comment]
Sorry to hear that, @sakshamcodes01. If you're still interested in working on this issue, there are plenty of Select components that need refactoring, so you and @besalraj30 can collaborate. Check out this PR – https://github.com/kiali/kiali/pull/7812 – as a reference for refactoring a Select component.

**Comment by sakshamcodes01** [comment]
Sure, @ferhoyos taking a look at the PR. Thanks!

**Comment by sakshamcodes01** [comment]
@besalraj30 If you're ok distributing the refactoring of Select components with me, can we connect on this?

**Comment by besalraj30** [comment]
@sakshamcodes01  sure. Sounds good. Even I have been pretty occupied. I'll share the distributed list and we can start working on it? Hope that's fine

**Comment by sakshamcodes01** [comment]
Sure thing! will be great if you can share with me the distributed list. @besalraj30

**Comment by jshaughn** [comment]
@sakshamcodes01 , @besalraj30 , any progress here?

**Comment by besalraj30** [comment]
Hi @sakshamcodes01 @ferhoyos @jshaughn,

Currently, FormSelect is being used in a total of 14 components, out of which changes have already been implemented in GatewaySelector and K8sGatewaySelector.

Here is the list of the remaining components:

1. TrafficPolicy
2. AuthorizationPolicyForm
3. K8sGatewayForm
4. K8sReferenceGrantForm
5. PeerAuthenticationForm
6. ServiceEntryForm
7. SourceBuilder
8. OperationBuilder
9. AddressBuilder
10. ListenerBuilder
11. ServerBuilder
12. JwtRuleBuilder

I will start working on the first six components (TrafficPolicy to ServiceEntryForm).
@sakshamcodes01 , could you take up the remaining ones (SourceBuilder to JwtRuleBuilder)?

Please update this thread with the components you’ve picked up and notify us when you raise a PR.

**Comment by sakshamcodes01** [comment]
@besalraj30 sure thing!

**Comment by sakshamcodes01** [comment]
@jshaughn @besalraj30 Hi! I have been trying to install kiali on my windows system but haven;t been able to. Iso tried WSL, but while running the [crc-openshift.sh](https://github.com/kiali/kiali/blob/master/hack/crc-openshift.sh) script, I was shown the error, that installation cannot proceed on WSL as it is unsupported as of now which was ultimately a dead end. Do you guys have any suggestions for windows install?

**Comment by jmazzitelli** [comment]
> @jshaughn @besalraj30 Hi! I have been trying to install kiali on my windows system but haven;t been able to. Iso tried WSL, but while running the crc-openshift.sh script, I was shown the error, that installation cannot proceed on WSL as it is unsupported as of now which was ultimately a dead end. Do you guys have any suggestions for windows install?

This question doesn't sound related to this issue at all. Sounds like you have a general discussion question - please submit your question here: https://github.com/kiali/kiali/discussions/categories/q-a (but I can tell you we most likely don't have any answers for getting things installed on Windows - I don't think we have anyone on the team or in the community doing that.)
